index.bs

```diff
@@ -1462,3 +1501,283 @@ The [=remote end steps=] with |command parameters| are:
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.contextCreated">
+When the [=create a new browsing context=] algorithm is invoked, after the
+[=active document=] of the browsing context is set, run the following steps:
+
+ 1. Let |context| be the newly created browsing context.
+
+ 1. Let |related contexts| be a set containing the [=parent browsing context=]
+    of |context|, if that is not null, or an empty set otherwise.
+
+ 1. Let |params| be the result of [=get the browsing context info=] given
+    |context|, 0, and 1.
+
+ 1. Let |body| be a [=map=] matching the
+    <code>BrowsingContextCreatedEvent</code> production, with the
+    <code>value</code> field set to |params|.
+
+ 1. [=Emit an event=] with |body| and |related contexts|.
+
+</div>
+
+#### The browsingContext.contextDestroyed Event #### {#event-browsingContext-contextDestroyed}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextDestroyedEvent = {
+         method: "browsingContext.contextDestroyed",
+         params: BrowsingContextDestroyedParameters
+       }
+
+       BrowsingContextDestroyedParameters = {
+         context: BrowsingContext,
+         parent: BrowsingContext / null,
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.contexDestroyed">
+
+Whenever the [=a browsing context is discarded=] algorithm is invoked, run the
+following steps after all {{Document}} objects have been discarded, but before
+the [=browsing context group is removed=] (if relevant):
+
+Note: The intent is that when a subtree of browsing contexts is removed, the
+<code>contextDestroyed</code> event is always emitted for a child before the
+parent.
+
+ 1. Let |context| be the browsing context being discarded.
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. If |context| has a [=parent browsing context=] let |parent| be that parent
+    browsing context and |parent id| be the [=browsing context id=] of
+    |parent|. Otherwise let |parent| and |parent id| be null.
+
+ 1. Let |params| be a [=map=] matching the <code>BrowsingContextDestroyedParameters</code>
+    production, with the <code>context</code> field set to |context id|,
+    and the <code>parent</code> field set to |parent id|.
+
+ 1. Let |body| be a [=map=] matching the
+    <code>BrowsingContextDestroyedEvent</code> production, with the
+    <code>params</code> field set to |params|.
+
+ 1. Let |related contexts| be a set, containing |parent| if that is not null, or
+    an empty set otherwise.
+
+ 1. [=Emit an event=] with |body| and |related contexts|.
+
+Issue: the way this hooks into HTML feels very fragile. See https://github.com/whatwg/html/issues/6194
+
+Issue: It's unclear if we ought to only fire this event for browsing
+contexts that have active documents; navigation can also cause contexts to
```

Comment:
CDP has a separate [`Page.frameNavigated`](https://chromedevtools.github.io/devtools-protocol/tot/Page/#event-frameNavigated) event. Navigating a browsing context will indeed result in the tree of browsing contexts changing, but I agree it's not clear how best to represent that.